src/io.c

```diff
@@ -1613,8 +1617,13 @@ static inline void FormatOutput(
       arg1 = arg2;
     }
 
-    /* '%s' print a string                                         */
-    else if ( *p == 's' ) {
+    /* '%s' or '%g' print a string                               */
+    else if ( *p == 's' || *p == 'g') {
+
+      /* If a GAP obj, get out the contained string              */
+      if(*p == 'g') {
+          arg1 = (Int)CSTR_STRING((Obj)arg1);
+      }
```

Comment:
Thinking about this some more, isn't this still unsafe, because `put_a_char` could still trigger a GC?

Reply:
Exactly, that's what @stevelinton said to me. Will come up with a fix.

Reply:
At least that's what I encountered when working on printing for integers. I ended up converting huge strings first into a string object, then calling `PrintString1` on it, see `PrintInt` `in src/integer.c`